h2o-core/src/main/java/hex/grid/Grid.java

```diff
@@ -540,6 +535,68 @@ public void exportModelsBinary(final String exportDir) throws IOException {
     }
   }
 
+  @SuppressWarnings("rawtypes")
+  private Set<Key> getDependentKeysFromParams() {
+    Field[] fields = Weaver.getWovenFields(getParams().getClass());
+    Set<Key> values = new HashSet<>();
+    for (Field f : fields) {
+      f.setAccessible(true);
+      Class<?> c = f.getType();
+      try {
+        Object value = f.get(getParams());
+        if (value instanceof Key) {
+          values.add((Key) value);
+        } else if (value != null && c.isArray() && c.getComponentType() == Key.class) {
+          Key[] arr = (Key[]) value;
+          for (Key k : arr) 
+            if (k != null) values.add(k);
+        }
+      } catch (IllegalAccessException e) {
+        throw new RuntimeException(e);
+      }
+    }
+    return values;
+  }
+  
+  private Job persistObj(
+      final Keyed o,
+      final String exportDir,
+      Map<String, String> referenceKeyTypeMap
+  ) {
+    if (o instanceof Frame) {
+      referenceKeyTypeMap.put(o._key.toString(), "frame");
+      return new FramePersist((Frame) o).saveTo(exportDir, true);
+    } else if (o != null) {
+      referenceKeyTypeMap.put(o._key.toString(), "iced");
+      URI dest = FileUtils.getURI(exportDir + "/" + o._key);
+      PersistUtils.write(dest, ab -> ab.putKey(o._key));
+    }
+    return null;
+  }
+
+  /**
+   * Saves all of the keyed objects used by this Grid's params. Files are named by objects' keys.
+   *
+   * @param exportDir directory to export all the frames to
+   */
+  public void exportReferencesBinary(final String exportDir) {
+    final Set<Key> keys = getDependentKeysFromParams();
+    final IcedHashMap<String, String> referenceKeyTypeMap = new IcedHashMap<>();
+    keys.stream()
+        .map(Key::get)
+        .map(obj -> persistObj(obj, exportDir, referenceKeyTypeMap))
```

Comment:
you keep exporting even after it already crashed because you are throwing an exception and skipping rest of the jobs - this behavior is definitely not ideal

simple sequential write would probably make more sense here

Reply:
fixed